src/coreclr/jit/valuenum.h

```diff
@@ -184,6 +184,43 @@ struct VNFuncApp
     }
 };
 
+class VNKindsIterator
```

Comment:
It is somewhat unfortunate, but MSVC unrolls loops better when using the iterator (as opposed to a static array that would been a simpler solution).

Reply:
Does it really matter for a debug check function? I would rather keep the source simple.

Reply:
Not for debug checks, but we have other places where it would be nice to use it (and I am planning to): `fgValueNumberAddExceptionSetForOverflow`, `fgValueNumberAddExceptionSetForDivision`, `fgValueNumberAddExceptionSetForIndirection`.

Reply:
Ok. I know this is a draft, but it would be nice to land all the updates together with its introduction so that we don't end up doing it in multiple different ways.

Reply:
Left `VNKinds` out of this change.